almende/vis
# [Proof of Concept] [Experimental] Adds automated visual regression testing using phantomjs and image ma…

…gick

Based on conversation on https://github.com/almende/vis/issues/2546 I've started this as a proof of concept for visual regression testing the static examples.

### What this currently does

1. Opens some examples on the visjs.org website and renders them as `png` files to the respective example directory.
2. Opens some examples on the `gh-pages` branch in git, and renders them as `png` files to the respective example directory.

<img width="206" alt="screen shot 2017-08-01 at 8 14 07 pm" src="https://user-images.githubusercontent.com/1480524/28852279-06d8609c-76f6-11e7-9698-3abc71bb2d40.png">

3. Does an image compare of one live example against one example from the `gh-pages` branch

I went with PhantomJS and ImageMagick, simply because they're already available in the standard Travis environment.

### To test this branch
- Install phantomjs
- Install ImageMagick
- Pull down this branch
- From the the vis directory:
    - run `phantomjs examplesToPng.js`
    - run `./magick.sh`
- <strong>Note the `-live.png` and `-local.png` examples in `examples/graph2/` </strong>
- <strong>Note that the result code on magick.sh will be 0 if the images match. </strong>

### Long-term intent
- Automate visual regression testing of all static examples
- Include all examples by default, with a way to opt-out
- Have visual regression tests on every PR, comparing the PR'd branch against the live page
- Fail the build if visual regression tests turn up bugs
